fairscale/nn/data_parallel/sharded_ddp.py

```diff
@@ -300,8 +290,8 @@ def reduce(self) -> None:
         # Trigger all the current BW hooks
         _ = map(lambda x: x(), self._grad_accs)
 
-        # Make sure that all the futures are consumed
-        self._consume_work_handles()
+        # Make sure that the buckets are flushed and all the futures are consumed
+        self._flush_reduce_calls()
```

Comment:
manual reduce case, spotted by the unit test, this was not flushing the buckets (so manual reduce + unused params would not be correct)